File: alliances/all-data/webapp/src/types/post.ts
import {Post as MattermostPost} from 'mattermost-webapp/packages/types/src/posts';

export type Post = {
    id: string;
    message: string;
};

// Maps a server-side Map with an object that has the map's keys as keys and map's values as values
export type IDMappedPosts = {
    [id: string]: MattermostPost;
}

export type PostsByIdsParams = {
    postIds: string[];
};

export type GetPostsByIdsResult = {
    posts: Post[];
};

export type PostsForTeamParams = {
    teamId: string;
};

export type GetPostsForTeamResult = {
    posts: IDMappedPosts;
};

File: alliances/all-data/server/app/post.go
package app

import "github.com/mattermost/mattermost-server/v6/model"

type Post struct {
	ID      string `json:"id"`
	Message string `json:"message"`
}

type PostsByIdsParams struct {
	PostIds []string `json:"postIds"`
}

type GetPostsByIdsResult struct {
	Posts []Post `json:"posts"`
}

type IDMappedPosts struct {
	Posts map[string]*model.Post `json:"posts"`
}
